int16 targetx
int16 targety
int8 doneshoot
int8 value
int8 robotStep
int8 umpan